rig-core/src/providers/perplexity.rs

```diff
@@ -227,25 +227,28 @@ impl completion::CompletionModel for CompletionModel {
         };
 
         // Add context documents to chat history
-        messages.append(
-            completion_request
-                .documents
-                .into_iter()
-                .map(|doc| completion::Message {
-                    role: "system".into(),
-                    content: serde_json::to_string(&doc).expect("Document should serialize"),
-                })
-                .collect::<Vec<_>>()
-                .as_mut(),
-        );
+        let prompt_content = if !completion_request.documents.is_empty() {
+            format!(
+                "<attachments>\n{}</attachments>\n\n{}",
+                completion_request
+                    .documents
+                    .iter()
+                    .map(|doc| doc.to_string())
+                    .collect::<Vec<_>>()
+                    .join("\n"),
+                completion_request.prompt
+            )
+        } else {
+            completion_request.prompt
+        };
```

Comment:
Since this is copied in many provider implementation, this can probably be refactored into it's own utility method `CompletionRequest::prompt_with_context(&self) -> String`